File: data/multilang_comparison/program_go/program_go/src/func29.go
package main
import "math/rand"
var _ = rand.Intn
func Func29(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var loop5 int = 0;
   var loopLimit5 int = (50)/4 + 1;
   for ; loop5 < loopLimit5; loop5++ {
      if (PATH0 & 1) != 0 {
         var array8 *Array
         if pCounter > 0 {
             array8 = vars.Data[pCounter-1]
             pCounter--
             array8.RefC++
         } else {
             array8 = &Array{
                 Size: 802,
                 RefC: 1,
                 Id: 8,
                 Data: make([]uint32, 802),
             }
         }
         var loop6 int = 0;
         var loopLimit6 int = (50)/5 + 1;
         for ; loop6 < loopLimit6; loop6++ {
            for i := 0; i < array8.Size; i++ {
                array8.Data[i]--
            }
            for i := 0; i < array8.Size; i++ {
                if array8.Data[i] == 69 {
                    return array8
                }
            }
         }
         var params0 ArrayParam
         params0.Size = 1
         params0.Data = make([]*Array, params0.Size)
         params0.Data[0] = array8
         var array9 *Array = Func46(&params0, getPath(), loopsFactor);
         params0.Data = nil
         params0.Size = 0
         array9.RefC--
         if array9.RefC == 0 {
             array9.Data = nil
             array9 = nil
         }
         array8.RefC--
         if array8.RefC == 0 {
             array8.Data = nil
             array8 = nil
         }
      
      } else {
         var params0 ArrayParam
         params0.Size = 0
         params0.Data = make([]*Array, params0.Size)
         var array16 *Array = Func40(&params0, loopsFactor);
         params0.Data = nil
         params0.Size = 0
         array16.RefC--
         if array16.RefC == 0 {
             array16.Data = nil
             array16 = nil
         }
      }
      var array18 *Array
      if pCounter > 0 {
          array18 = vars.Data[pCounter-1]
          pCounter--
          array18.RefC++
      } else {
          array18 = &Array{
              Size: 857,
              RefC: 1,
              Id: 18,
              Data: make([]uint32, 857),
          }
      }
      var params0 ArrayParam
      params0.Size = 1
      params0.Data = make([]*Array, params0.Size)
      params0.Data[0] = array18
      var array19 *Array = Func41(&params0, loopsFactor);
      params0.Data = nil
      params0.Size = 0
      if (PATH0 & 2) != 0 {
         var array22 *Array
         if pCounter > 0 {
             array22 = vars.Data[pCounter-1]
             pCounter--
             array22.RefC++
         } else {
             array22 = &Array{
                 Size: 808,
                 RefC: 1,
                 Id: 22,
                 Data: make([]uint32, 808),
             }
         }
         var loop9 int = 0;
         var loopLimit9 int = (50)/5 + 1;
         for ; loop9 < loopLimit9; loop9++ {
            for i := 0; i < array18.Size; i++ {
                array18.Data[i]--
            }
            for i := 0; i < array22.Size; i++ {
                if array22.Data[i] == 88 {
                    return array22
                }
            }
         }
         array22.RefC--
         if array22.RefC == 0 {
             array22.Data = nil
             array22 = nil
         }
      
      } else {
         var params1 ArrayParam
         params1.Size = 2
         params1.Data = make([]*Array, params1.Size)
         params1.Data[0] = array18
         params1.Data[1] = array19
         var array23 *Array = Func50(&params1, loopsFactor);
         params1.Data = nil
         params1.Size = 0
         array23.RefC--
         if array23.RefC == 0 {
             array23.Data = nil
             array23 = nil
         }
      }
      for i := 0; i < array19.Size; i++ {
          if array19.Data[i] == 12 {
              return array19
          }
      }
      array19.RefC--
      if array19.RefC == 0 {
          array19.Data = nil
          array19 = nil
      }
      array18.RefC--
      if array18.RefC == 0 {
          array18.Data = nil
          array18 = nil
      }
   }
   var array25 *Array
   if pCounter > 0 {
       array25 = vars.Data[pCounter-1]
       pCounter--
       array25.RefC++
   } else {
       array25 = &Array{
           Size: 226,
           RefC: 1,
           Id: 25,
           Data: make([]uint32, 226),
       }
   }
   return array25;
}

File: data/multilang_comparison/program_julia/program_julia/src/func29.jl
include("program_julia_head.jl")
function func29(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   loopLimit5 = (50)/4 + 1
   for i in 1:loopLimit5 
      if (PATH0 & 1) != 0
         array8 = nothing
         if pCounter > 1
             array8 = vars.data[pCounter]
             pCounter -= 1
             array8.refC += 1
         else
             array8 = JArray(zeros(Int, 802), 802, 1, 8)
         end
         loopLimit6 = (50)/5 + 1
         for i in 1:loopLimit6 
            for i in 0:(array8.size-1) 
               array8.data[i+1] -= 1
            end
            for i in 1:array8.size
                if array8.data[i] == 69
                        return array8
               end
            end
         end
         params0 = JArray_param{Int}()
         params0.size = 1
         push!(params0.data, array8)
         params0.size += 1
         array9 = func46(params0, get_path(), loopsFactor);
         array9.refC -= 1
         if array9.refC == 0
             empty!(array9.data)
             array9 = nothing
         end
         array8.refC -= 1
         if array8.refC == 0
             empty!(array8.data)
             array8 = nothing
         end
      
      else 
         params0 = JArray_param{Int}()
         params0.size = 0
         array16 = func40(params0, loopsFactor);
         array16.refC -= 1
         if array16.refC == 0
             empty!(array16.data)
             array16 = nothing
         end
      end
      array18 = nothing
      if pCounter > 1
          array18 = vars.data[pCounter]
          pCounter -= 1
          array18.refC += 1
      else
          array18 = JArray(zeros(Int, 857), 857, 1, 18)
      end
      params0 = JArray_param{Int}()
      params0.size = 1
      push!(params0.data, array18)
      params0.size += 1
      array19 = func41(params0, loopsFactor);
      if (PATH0 & 2) != 0
         array22 = nothing
         if pCounter > 1
             array22 = vars.data[pCounter]
             pCounter -= 1
             array22.refC += 1
         else
             array22 = JArray(zeros(Int, 808), 808, 1, 22)
         end
         loopLimit9 = (50)/5 + 1
         for i in 1:loopLimit9 
            for i in 0:(array18.size-1) 
               array18.data[i+1] -= 1
            end
            for i in 1:array22.size
                if array22.data[i] == 88
                        return array22
               end
            end
         end
         array22.refC -= 1
         if array22.refC == 0
             empty!(array22.data)
             array22 = nothing
         end
      
      else 
         params1 = JArray_param{Int}()
         params1.size = 2
         push!(params1.data, array18)
         params1.size += 1
         push!(params1.data, array19)
         params1.size += 1
         array23 = func50(params1, loopsFactor);
         array23.refC -= 1
         if array23.refC == 0
             empty!(array23.data)
             array23 = nothing
         end
      end
      for i in 1:array19.size
          if array19.data[i] == 12
                  return array19
         end
      end
      array19.refC -= 1
      if array19.refC == 0
          empty!(array19.data)
          array19 = nothing
      end
      array18.refC -= 1
      if array18.refC == 0
          empty!(array18.data)
          array18 = nothing
      end
   end
   array25 = nothing
   if pCounter > 1
       array25 = vars.data[pCounter]
       pCounter -= 1
       array25.refC += 1
   else
       array25 = JArray(zeros(Int, 226), 226, 1, 25)
   end
   return array25;
end

